sdk/python/testvectors/__main__.py

```diff
@@ -187,51 +315,72 @@ def create_entry(self, schema_name, test_name, factory, descriptor):
 			'descriptor': self.fix_descriptor_before_storing(final_descriptor)
 		}
 
-	def create_transactions(self, module_descriptor, recipes):
+	def create_objects(self, module_descriptor, recipes):
 		test_cases = []
-		schema_name = recipes['schema_name']
-		test_prefix = f'{schema_name}_{module_descriptor[0]}'
-		for index, descriptor in enumerate(recipes['descriptors']):
-			test_name = f'{test_prefix}_single_{index+1}'
-			test_cases.append(self.create_entry(schema_name, test_name, self.helper.create, descriptor))
-
-		# only thing _not_ supporting wrapping in aggregates is NEM's Cosignature (transaction)
-		if recipes.get('single_only', False):
-			return test_cases
 
-		for index, descriptor in enumerate(recipes['descriptors']):
-			test_name = f'{test_prefix}_aggregate_{index+1}'
-			aggregate_schema_name = self.helper.AGGREGATE_SCHEMA_NAME
-			test_cases.append(self.create_entry(aggregate_schema_name, test_name, self.helper.create_aggregate_from_single, descriptor))
+		had_transaction = False
+		for index, entry in enumerate(recipes):
+			schema_name = entry['schema_name']
+			test_prefix = f'{schema_name}_{module_descriptor[0]}'
 
-		return test_cases
+			object_type = self.helper.determine_type(entry['descriptor'])
 
-	def create_aggregate_transactions(self, module_descriptor, recipes):
-		test_cases = []
-		schema_name = recipes['schema_name']
-		test_prefix = f'{schema_name}_{module_descriptor[0]}'
+			if 'transaction' == object_type:
+				had_transaction = True
+
+			test_name = self.helper.get_test_name(object_type, test_prefix, index)
+			handler = self.helper.get_handler(object_type)
+
+			test_cases.append(self.create_entry(schema_name, test_name, handler, entry['descriptor']))
+
+		if not had_transaction:
+			return test_cases
+
+		# if there was a transaction, need to run loop once more to generate that transaction wrapped in an aggregate transaction
+		for index, entry in enumerate(recipes):
+			if 'transaction' != self.helper.determine_type(entry['descriptor']):
+				continue
 
-		for index, descriptor in enumerate(recipes['descriptors']):
+			# only thing _not_ supporting wrapping in aggregates is NEM's Cosignature (transaction)
+			if entry.get('single_only', False):
+				continue
+
+			test_prefix = f'{entry["schema_name"]}_{module_descriptor[0]}'
 			test_name = f'{test_prefix}_aggregate_{index+1}'
```

Comment:
get_test_name ?

Reply:
no, this is for a case where:
 a) we're generating transaction (above)
 b) here we're generating same transaction wrapped inside an aggregate (so it has different test name + signer etc)
 
 sample:
 ```json
     "test_name": "AddressAliasTransactionV1_address_alias_single_2",
...
      "signer_public_key": "61C09C365DE9685B3E93420019F9F4BC853013E1AD95C67E7BBA32DB6C44D1C9",
...
    "test_name": "AddressAliasTransactionV1_address_alias_aggregate_2",
...
      "signer_public_key": "9EB78F164D02FE97FE29259E2678586A7BC08E402D14C1444A64A97B394D1F48",
 ```